spec/units/tab_completion/dsl/parser_spec.rb

```diff
@@ -0,0 +1,133 @@
+require 'spec_helper'
+require 'gitsh/tab_completion/dsl/parser'
+
+describe Gitsh::TabCompletion::DSL::Parser do
+  describe '.parse' do
+    it 'parses single words' do
+      result = parse_single_rule(tokens([:WORD, 'stash'], [:EOS]))
+
+      expect(result).to be_a_text_transition
+      expect(result.word).to eq('stash')
+    end
+
+    it 'parses single variables' do
+      result = parse_single_rule(tokens([:VAR, 'revision'], [:EOS]))
+
+      expect(result).to be_a_variable_transition
+      expect(result.matcher).to be_a_revision_matcher
+    end
+
+    it 'parses rules with multiple words and variables' do
+      result = parse_single_rule(tokens(
+        [:WORD, 'stash'], [:WORD, 'pop'], [:VAR, 'revision'], [:EOS],
```

Comment:
The indentation comment seems to be an example of this issue, so I guess we'll just live with that one: https://github.com/bbatsov/rubocop/issues/4308